test/sqllogictest/arrays.slt

```diff
@@ -306,9 +306,10 @@ SELECT ARRAY[1, 2, 3][1.5 + 1.5]
 ----
 3
 
-# Error different than Cockroach.
-query error cannot subscript type text
+query T
 SELECT ARRAY['a', 'b', 'c'][4][2]
+----
+NULL
```

Comment:
Unless I missed something, I didn't see a test in this file that tested what happens with NULL in the subscript place. Do we need to add some of those?

Reply:
Big derp on my end; will add those tests/handling throughout.

Reply:
Ah, this is automatically/already handled:

https://github.com/MaterializeInc/materialize/blob/main/test/sqllogictest/list.slt#L430
https://github.com/MaterializeInc/materialize/blob/main/test/sqllogictest/list.slt#L435

Reply:
:+1: